tt_metal/impl/debug/watcher_server.cpp

```diff
@@ -246,6 +266,58 @@ static void dump_noc_sanity_status(FILE *f,
     }
 }
 
+static void dump_assert_status(
+    FILE *f,
+    CoreCoord core,
+    const launch_msg_t *launch_msg,
+    const debug_assert_msg_t *assert_status,
+    const debug_status_msg_t *debug_status
+) {
+    switch (assert_status->tripped) {
+        case DebugAssertTripped: {
+            // TODO: Get rid of this once #6098 is implemented.
+            std::string line_num_warning = "Note that file name reporting is not yet implemented, and the reported line number for the assert may be from a different file.";
+            fprintf(
+                f, "%s tripped assert on line %d. Current kernel: %s. %s",
+                get_riscv_name(core, launch_msg, assert_status->which),
+                assert_status->line_num,
+                get_kernel_name(core, launch_msg, assert_status->which).c_str(),
+                line_num_warning.c_str()
+            );
+            fflush(f);
+            log_running_kernels(launch_msg);
+            log_waypoint(core, launch_msg, debug_status);
+            log_info(LogLLRuntime, "Watcher stopped the device due to tripped assert.");
+            TT_THROW(
+                "Watcher detected an assert: core {}, riscv {}, line {}. Current kernel: {}. {}",
+                core.str(),
+                get_riscv_name(core, launch_msg, assert_status->which),
+                assert_status->line_num,
+                get_kernel_name(core, launch_msg, assert_status->which),
+                line_num_warning
+            );
+            break;
+        }
+        case DebugAssertOK:
+            if (assert_status->line_num != DEBUG_SANITIZE_NOC_SENTINEL_OK_16 ||
+                assert_status->which != DEBUG_SANITIZE_NOC_SENTINEL_OK_8) {
+                TT_THROW(
```

Comment:
looks like this does go to stderr?

Reply:
Yes, goes to stderr and the watcher log, corrected documentation